tutor/hooks/actions.py

```diff
@@ -4,58 +4,83 @@
 import sys
 import typing as t
 
+from typing_extensions import ParamSpec
+
 from .contexts import Contextualized
 
-# Similarly to CallableFilter, it should be possible to refine the definition of
-# CallableAction in the future.
-CallableAction = t.Callable[..., None]
+P = ParamSpec("P")
+# Similarly to CallableFilter, it should be possible to create a CallableAction alias in
+# the future.
```

Comment:
nit: If "in the future" means "after Python 3.X is end-of-life", and if you know what X is, then you might just note that here for future developers.

Reply:
It's actually independent of Python version, and only once the corresponding mypy issue (documented in filters.py) will be resolved.